testutils/src/main/scala/za/co/absa/enceladus/testutils/e2eSparkRunner/CmdConfig.scala

```diff
@@ -0,0 +1,220 @@
+                     reportVersion: String = "",
+                     sparkConfFile: String = "") {
+  private val stdPaths = "conformance-output/standardized-" +
+                        s"$datasetName-" +
+                        s"$datasetVersion-" +
+                        s"$reportDate-" +
+                        s"$reportVersion"
+  private val confPaths = s"$datasetName/" +
+                          s"enceladus_info_date=$reportDate/" +
+                          s"enceladus_info_version=$reportVersion"
+
+  private val defaultStdOut = s"/tmp/$stdPaths"
+  private val defaultConfOut = s"/publish/$confPaths"
+
+  private val refStdOut = s"/ref/tmp/$stdPaths"
+  private val refConfOut = s"/ref/publish/$confPaths"
+
+  private val cmpStdPath = s"/cmp/tmp/$stdPaths"
+  private val cmpConfPath = s"/cmp/publish/$confPaths"
+
+  val sparkConf: String = Source.fromFile(sparkConfFile).getLines().mkString
+
+  val stdComparisonConf: String = comparisonConf(defaultStdOut, refStdOut, cmpStdPath, keys)
+  val confComparisonConf: String = comparisonConf(defaultConfOut, refConfOut, cmpConfPath, keys)
+
+  /**
+    * Get Conformance options like dataset name, dataset version, etc.
+    * @return A string of conformance job options
+    */
+  def conformanceJobConf: String = {
+    s"--dataset-name $datasetName " +
+      s"--dataset-version $datasetVersion " +
+      s"--report-date $reportDate " +
+      s"--report-version $reportVersion " +
+      s"--menas-credentials-file $menasCredentialsFile"
+  }
+
+  /**
+    * Get Standartization options like dataset name, dataset version, etc.
+    * @return A string of Standartization job options
+    */
+  def standartizationJobConf: String = {
+    val base: String = s"--dataset-name $datasetName " +
+      s"--dataset-version $datasetVersion " +
+      s"--report-date $reportDate " +
+      s"--report-version $reportVersion " +
+      s"--menas-credentials-file $menasCredentialsFile"
+
+    val format: String = rawFormat match {
+      case "csv" => s"--raw-format $rawFormat --delimiter ${csvDelimiter.get} --header ${csvHeader.get}"
+      case "xml" => s"--raw-format $rawFormat --row-tag ${rowTag.get}"
+      case _ => s"--raw-format $rawFormat"
+    }
+
+    s"$base $format"
+  }
+
+  private def comparisonConf(newPath: String, refPath: String, outPath: String, keys: String): String = {
+    s"--raw-format parquet --new-path $newPath --ref-path $refPath --out-path $outPath --keys $keys"
+  }
+}
+
+object CmdConfig {
+
+  /**
+    * Parses and validates an Array of input parameters and creates an instance of CmdConfig case class
+    * @param args Array of argument to be parsed
+    * @return Returns a CmdConfig instance holding parameters passed
+    */
+  def getCmdLineArguments(args: Array[String]): CmdConfig = {
+    val parser = new CmdParser("spark-submit [spark options] TestUtils.jar")
+
+    val optionCmd = parser.parse(args, CmdConfig())
+    if (optionCmd.isEmpty) {
```

Comment:
This is another good candidate for `.get` replacement with pattern matching.

Reply:
I will go ahead and mark this as resolved.